marketdata/charts.go

```diff
@@ -52,8 +53,23 @@ func (c *Charts) GetCoinInfo(coin uint, token string, currency string) (blockatl
 	return coinInfoData, errors.E("No chart coin info data found", errors.Params{"coin": coin, "token": token})
 }
 
-func normalizePrices(prices []blockatlas.ChartPrice) {
+func normalizePrices(prices []blockatlas.ChartPrice, maxItems int) (result []blockatlas.ChartPrice) {
 	sort.Slice(prices, func(p, q int) bool {
 		return prices[p].Date < prices[q].Date
 	})
+	if len(prices) > maxItems && maxItems > 0 {
+		skip := int(math.Ceil(float64(len(prices) / maxItems)))
+		i := 0
+		for i < len(prices) {
+			result = append(result, prices[i])
+			i += skip + 1
+		}
+		lastPrice := prices[len(prices)-1]
+		if len(result) > 0 && lastPrice.Date != result[len(result)-1].Date {
+			result = append(result, lastPrice)
+		}
```

Comment:
```suggestion
		for i := 0; i < len(prices) || i == len(result)-1; i += skip + 1 {
			result = append(result, prices[i])
		}
```

Reply:
this solution might not work since skip may overcome len(prices) and be bigger. Hence condition `i == len(result)-1` won't take place.

Reply:
But you can remove the lines 68 and 71 in one simple for interaction

```
		for i := 0; i < len(prices); i += skip + 1 {
			result = append(result, prices[i])
		}
```